packages/core/admin/admin/src/content-manager/hooks/useLazyComponents/index.js

```diff
@@ -1,44 +1,69 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { useCustomFields } from '@strapi/helper-plugin';
 
+const componentStore = new Map();
+
 /**
  * @description
  * A hook to lazy load custom field components
  * @param {Array.<string>} componentUids - The uids to look up components
  * @returns object
  */
-const useLazyComponents = (componentUids) => {
-  const [lazyComponentStore, setLazyComponentStore] = useState({});
-  const [loading, setLoading] = useState(true);
+const useLazyComponents = (componentUids = []) => {
+  const [lazyComponentStore, setLazyComponentStore] = useState(Object.fromEntries(componentStore));
+  const [loading, setLoading] = useState(() => {
+    if (componentStore.size === 0 && componentUids.length > 0) {
+      return true;
+    }
```

Comment:
@joshuaellis Just wondering: would this create problems for users who don't have a component defined yet at all? Would this result in an infinite `loading` state?

Reply:
I don't think so, the check here is "is there an empty cache and do we have a list of components to fetch", so someone typically has to be passing inputs to be fetched for loading to be default true.

There is a test around this, I think we're okay?